### ISSUE TITLE
[FEAT] Make implementing "save and continue editing" button easier

### ISSUE THREAD

**Issue by rassie** [pankod/refine#1617]
See discussion on [Discord](https://discord.com/channels/837692625737613362/950759862114074674)

**Is your feature request related to a problem? Please describe.**
Django-Admin implements two buttons on its edit pages: "Save" and "Save and continue editing". The latter is sorely missed in Refine.

**Describe the solution you'd like**
As discussed on Discord, a possible solution would be to have `useForm` return a mutation function, which can be combined with `{ redirect: false }` option and `actionButtons` on `<Edit>` to build both buttons.

**Comment by omeraplak** [comment]
Hey @rassie ,
Thank you! Let's see what we can do

**Comment by omeraplak** [comment]
Hey @rassie ,
I have an idea. For "pessimistic" mode we will make `onFinish` Async so you can "await" the result.

How does this sound?
```
const { form, onFinish } = useForm<IPost>({
             redirect: false,
         });

actionButtons={
                 <>
                     <SaveButton
                         onClick={async() => {
                             await onFinish?.(form.getFieldsValue(true));
                         }}
                     >
                         Save and continue editing
                     </SaveButton>
                     <SaveButton
                         onClick={async() => {
                             await onFinish?.(form.getFieldsValue(true));
                             list("posts");
                         }}
                     />
                 </>
             }
```

**Comment by rassie** [comment]
I think this looks good for a simple solution. But I also feel this can be simplified a bit, for better DX.

`list("posts")` is basically what `useForm({ redirect: "list" })` is doing, but with an additional burden to explicitly name the resource in question. Yes, we are writing code for a particular resource, but to have to type its name only a couple of times in code is a feature not a bug. Also, `form.getFieldsValue(true)` could probably be a default argument value, since both `onFinish` and `form` come from `useForm`, i.e. are available in the same scope. Then we'd arrive at this:

```
const { form, onFinish, redirect } = useForm<IPost>({
             automaticRedirect: false,
         });

actionButtons={
                 <>
                     <SaveButton
                         onClick={async() => {
                             await onFinish?.();
                         }}
                     >
                         Save and continue editing
                     </SaveButton>
                     <SaveButton
                         onClick={() => {
                             onFinish?.().then(redirect);
                         }}
                     />
                 </>
             }
```

But then again, `<SaveButton>` is a Refine component, so if we use all defaults, we might be talking about 

```
const { form } = useForm<IPost>();

actionButtons={
                 <>
                     <SaveButton redirect={false}>
                         Save and continue editing
                     </SaveButton>
                     <SaveButton />
                 </>
             }
```

but I'm not sure whether this would be easy to implement, since `<SaveButton>` doesn't handle redirect and probably shouldn't.

**Comment by omeraplak** [comment]
Hey @rassie ,
Your suggestion is much better. Thank you so much for always taking us forward!

**Comment by omeraplak** [comment]
Hey @rassie ,
I hope you are very well. I think We have a good start. Thank you again! 

For optimistic and undoable mutation modes, redirection is a bit problematic if "warnWhenUnsavedChanges" is active. We will solve this.

I'll discuss this approach with the team tomorrow. How is it for you for now?

Thanks!

**Comment by omeraplak** [comment]
Hey @rassie ,
I think it's ready for your review. We have updated the doc. What do you think?
https://save-an-doc-refine.pankod.com/docs/ui-frameworks/antd/hooks/form/useForm/#how-can-i-implement-the-save-and-continue-feature

**Comment by rassie** [comment]
From the look at the docs, I think it looks good, it does what it's supposed to do. One thing I'm confused about is 
```
                    <SaveButton {...saveButtonProps} onClick={() => onFinish()}>
                        Save and continue editing
                    </SaveButton>
```
Isn't `onFinish` the default for `onClick` anyway?

**Comment by omeraplak** [comment]
> 

Ohh, thank you. I've fixed it!

**Comment by omeraplak** [comment]
Ohh, thank you. I've fixed it!

**Comment by omeraplak** [comment]
Hey @rassie ,
We've released [`3.8.0`](https://github.com/pankod/refine/releases/tag/v3.8.0). Thank you!
